snmp/datadog_checks/snmp/snmp.py

```diff
@@ -246,11 +250,13 @@ def fetch_oids(self, config, oids, enforce_constraints):
                     self.log.debug(
                         'Running SNMP command getNext on OIDS: %s', OIDPrinter(missing_results, with_values=False)
                     )
-                    binds = snmp_getnext(
-                        config,
-                        missing_results,
-                        lookup_mib=enforce_constraints,
-                        ignore_nonincreasing_oid=self.ignore_nonincreasing_oid,
+                    binds = list(
```

Comment:
The `list()` usage here is actually a bug fix: since `snmp_getnext()` returns a generator, when in debug mode `OIDPrinter()` would consume the generator and `all_binds.extend(binds)` would fail with "generator already consumed".